content/post/codesent.md

```diff
@@ -0,0 +1,396 @@
+```go
+// It is low-level, old, and unused by Go's current HTTP stack
+
+// not clear why this shouldn't work
+
+// overflow causes something awful
+
+// quiet expected TLS handshake error remote error: bad certificate
+```
+Source: [1](https://github.com/golang/go/blob/master/src/net/http/httputil/persist.go#L33),
+[2](https://github.com/golang/go/blob/32aa0d9198a1b147eb2c756c0d5628b3399f9898/test/bugs/bug046.go#L11),
+[3](https://github.com/golang/go/blob/master/src/runtime/vlrt.go#L95),
+[4](https://github.com/golang/go/blob/master/src/os/exec/exec_test.go#L568).
+
+Positive:
+
+```go
+// Special case, useful in debugging
+
+// Implementation: Parallel summing of adjacent bits. See Hackers Delight, Chap. 5: Counting Bits.
+
+// TODO(austin): This could be a really awesome string method
+
+// TODO(bradfitz): this might be 0, once escape analysis is better
+```
+
+Source: [1](https://github.com/golang/go/blob/master/src/bytes/buffer.go#L63),
+[2](https://github.com/golang/go/blob/master/src/math/bits/bits.go#L135),
+[3](https://github.com/golang/go/blob/46e392e01c630dee41a67e01223b538aae9dc9b5/usr/austin/ogle/vars.go#L42),
+[4](https://github.com/golang/go/blob/84ef97b59c89b7d9fdc04a1a8a438cd3257bf521/src/pkg/strconv/strconv_test.go#L21).
+
+The last one is actually written by Brad who gave the Gophercon talk right before me hehe. In this case the model
+must have caught word "better".
+
+Finally, here are the positive comments in Keras, the deep learning framework I used to train
+BiDiSentiment model:
+
+```python
+# With TensorFlow, we can infer the output shape directly
+
+# Make sure there is exactly one 1 in a row
+
+# get sorted list of layer depths
+
+# Theano has a built-in optimization for logsumexp so we can just write the expression directly
+```
+
+I have looked at other classification results and can conclude the following. First, the model
+was trained on data of one nature (tweets) and applied to data of a different nature (comments).
+Thus it often exaggerates comment sentiment. Second, it behaves differently on two project types.
+The first type is projects which contain many details, many conditions, many checks and
+handle many edge cases. For example, language compilers (Go) or web servers (Django). BiDiSentiment
+classifies them as negative: the text tells how **not** to fail.
+The second type is projects which explain what is happening, which environment features they use
+to do better or explain much domain terminology. For example, k8s has tons of devops comments
+and Keras describes deep learning recipes. BiDiSentiment treats them as positive: they teach or
+explain why they are cool. I must say that deep learning seems to be the modern alchemy
+([discussion](https://medium.com/@Synced/lecun-vs-rahimi-has-machine-learning-become-alchemy-21cb1557920d))
+so Keras has a big positive sentiment, hehe.
+
+## Conclusion
+
+* It is easy to analyse Git repositories with Go.
+* Go is handy for inference-in-a-box with Tensorflow.
```

Comment:
`inference-in-a-box`
is this a well known term?
Maybe something like: Go provides an easy deployment of tools performing inference on Tensorflow models.

Reply:
This will be the name I will do my best to push to the community